src/Deserialization/images.jl

```diff
@@ -14,7 +14,7 @@ function lookahead_deserialize_image_summary(old_tag, old_val, evs::Summary,
 
     state = state_old + 1
     iter_result = iterate(evs, state)
-    while iter_result !== nothing
+    while !(iter_result isa Nothing)
```

Comment:
The previous code was fine.
`!==` is that same as `!(iter_result === nothing)` 
which (unlike `!=`) the compiler can optimize easily based on type information.
Since `===` can only be true of things with same type, and since `Nothing` is a singleton type


See demonstration:
```julia
julia> foo(x) = x!==nothing
foo (generic function with 1 method)

julia> bar(x) = !(x isa Nothing)
bar (generic function with 1 method)

julia> @code_llvm foo(1)
;  @ REPL[1]:1 within `foo`
define i8 @julia_foo_133(i64 signext %0) #0 {
top:
  ret i8 1
}

julia> @code_llvm foo(nothing)
;  @ REPL[1]:1 within `foo`
define i8 @julia_foo_151() #0 {
top:
  ret i8 0
}

julia> @code_llvm bar(1)
;  @ REPL[2]:1 within `bar`
define i8 @julia_bar_156(i64 signext %0) #0 {
top:
  ret i8 1
}

julia> @code_llvm bar(nothing)
;  @ REPL[2]:1 within `bar`
define i8 @julia_bar_158() #0 {
top:
  ret i8 0
}
```
either way it constant folds

Reply:
```suggestion
    while iter_result !== nothing
```